libsolidity/formal/CHC.cpp

```diff
@@ -642,10 +642,20 @@ void CHC::externalFunctionCall(FunctionCall const& _funCall)
 	}
 
 	auto postCallState = vector<smtutil::Expression>{state().state()} + currentStateVariables();
-	auto nondet = (*m_nondetInterfaces.at(m_currentContract))(preCallState + postCallState);
+	auto error = errorFlag().increaseIndex();
+	auto nondet = (*m_nondetInterfaces.at(m_currentContract))(preCallState + postCallState + vector<smtutil::Expression>{error, state().thisAddress()});
 	// TODO this could instead add the summary of the called function, where that summary
 	// basically has the nondet interface of this summary as a constraint.
```

Comment:
Does this TODO still make sense?
It seems outdated.

Reply:
I think I wrote that thinking about the counterexamples containing `nondet_interface`. Currently, since we add `nondet_interface` directly, that's all we see in the counterexample graph, and can't really synthesize what the external function did, in case we have more than one being called inside the same analyzed function. If we'd do that indirectly we might get more information.
Example:

```
contract C {
    uint x;
    function inc() public { require(x < 1000); ++x; }
    function f(D d) public {
        x = 50;
        d.g();
        d.h();
        assert(x <= 60);
    }
}
```

Of course a silly example, but: the assertion fails. For that, `d.g()` and `d.h()`, combined, would have to call (direct or indirectly) `inc()` at least 10 times. Since both calls just add `nondet_interface` constraints, we don't know which called what (inc) how many times. This is not important for this case, but it might in some other cases.
If we create summaries for `summary_D_g_untrusted` and `summary_D_h_untrusted` which use `nondet_interface`, we might be able to retrieve the information of which of those called what.

Reply:
I think that information should be present in the CEX already. Maybe we should discuss this in a call.

Reply:
The result of the discussion is now documented in https://github.com/ethereum/solidity/issues/9393
The TODO still makes sense.